### PR TITLE
[WIP] Allow different types for user-defined properties

### PR DESCRIPTION
**Pull request by mvz**

### CODE REVIEW

File: lib/ffi-gobject/param_spec.rb [hunk 1]
```diff
@@ -12,5 +12,28 @@ def ref
...
+      when TYPE_STRING
+        true
+      end
+    end
+
+    G_TYPE_MAP = {
+      TYPE_INT => :gint,
+      TYPE_STRING => :utf8,
+      TYPE_LONG => :glong
+    }.freeze
```
Comment (kugel-) [inline lib/ffi-gobject/param_spec.rb]: The map seems to fail for unhandled types. I've got a GObject property and a non-fundamental (but GBoxed-derived) types which don't work with it 
```
/home/kugel/.gem/ruby/2.3.0/gems/gir_ffi-0.10.2/lib/ffi-gobject/param_spec.rb:35:in `fetch': key not found: 19016480 (KeyError)
	from /home/kugel/.gem/ruby/2.3.0/gems/gir_ffi-0.10.2/lib/ffi-gobject/param_spec.rb:35:in `type_tag'
	from /home/kugel/.gem/ruby/2.3.0/gems/gir_ffi-0.10.2/lib/ffi-gobject/param_spec.rb:39:in `ffi_type'
```

Comment (mvz) [inline lib/ffi-gobject/param_spec.rb]: Yes, I still need to add other types and properly handle non-fundamental types.

**Comment by kugel-** [comment]
Seems to work great for my use case. Is there anything else I can check?

Thanks a lot, btw!

**Comment by kugel-** [comment]
:+1:

**Pull request merged**